plugin/Settings.pm

```diff
@@ -31,65 +31,71 @@ sub handler {
 	
 	$log->debug("Groups::Settings->handler() called.");
 	
-	if ($params->{'saveSettings'}) {
+	if ($params->{saveSettings}) {
+		my $groups = $prefs->get('groups') || [];
```

Comment:
now I'm more confused. Why keeping a group item in the prefs that simply lists the groups? Now that you have clients, isn't that redundant and risk of de-sync? Why not building the list on demand from getting the list of clients from the prefs

Reply:
something like keys $prefs->{clients} - doesn't that creates on-the-fly the list of groups?

Reply:
okay, so this $prefs->{clients} is a "cache" and it only works if the prefs has been read at least once, for all id's. So that worked in Settings.pm only because the initPluin reads the prefs of each client - isn't that a bit risky? If for one reason I delay the creation of players, the the Settings.pm will not work any more

Reply:
Preferences are kept in memory indeed, read from disk whenever preferences('namespace') is called first. Therefore it's reasonable to always read the prefs, rather than keeping (another) copy around.

Good catch about the redundancy of the groups pref. I thought it wasn't possible to access client prefs without the clients being connected, without digging too much in the prefs structure. But there is an allClients() call which would give me direct access. I didn't remember it, though I've been using it in my own plugins myself... New pull request is coming to "fix" this.